File: AS-2023-2024/5E-TPSIT/es-03_2023-10-19_general_java_OOP/src/main/java/model/Student.java
package model;
public class Student extends Person {
	private String classroom;
	public Student(String firstname, String lastname, int age, String classroom) {
		super(firstname, lastname, age);
		this.classroom = classroom != null && classroom.strip().length() == 2 ? classroom.strip() : "6Z";
	}
	public String getClassroom() {
		return classroom;
	}
	public void setClassroom(String classroom) {
		if(!classroom.isBlank() && classroom.strip().length() == 2)
			this.classroom = classroom.strip();
	}
	@Override
	public String toString() {
		return "%s\nI'm in the %s class!"
			.formatted(
				super.toString(), classroom
			);
	}
}

File: AS-2022-2023/5X-TPSIT/es-04/Student.py
"""
Python, like C ++, is a multiple inheritance language. To inherit a class, simply open the brackets and write the name of the superclass.
Example
class A (B):
it means that A inherits from B
"""

from Person import Person

class Student(Person):
	'Student template class'

	def __init__(
		self,
		firstname = '', lastname = '', age = -1,
		school = ''
	):
		# super class
		super().__init__(firstname, lastname, age)
		# private method
		self.__school = 'ITI Renato Elia' if not school.strip() else school.strip()

# to string
	def __str__(self):
		return f"{super().__str__()}. I'm study at {self.__school}"

# getter / setter
	def get_school(self):
		return school

	def set_school(self, school):
		self.__school = school.strip() if school.strip() else self.__school